### PR TITLE
Cosmic's Golf Course

### PR DESCRIPTION
**Pull request by cosmiccoincidence**
Adds..
-Golfcourse (map + overmap special + overmap terrain)
-golf gloves (item)
-golf hat (item)
-golf bag (item)
-golf shoes (item)
-golf tee (item)
-golf ball (item)
-above items to various item groups
-golf cart (item group)
-golf green (item group)
-long grass (terrain)
-dead grass (terrain)
-golf green (terrain)
-golf hole (terrain)
-golf cart 4 seat varient (vehicle)
-golf carts (vehicle group)

### CODE REVIEW

**Comment by Xhuis** [comment]
Might not be a bad idea to add some of the clothes to clothing stores and sporting goods stores. Just s suggestion, though.

**Comment by John-Candlebury** [comment]
You should make the golfbag be a scabbard for golfing irons!

Fore!

File: data/json/item_groups.json [hunk 1]
```diff
@@ -208,6 +208,12 @@
       ["fan", 10],
       ["pipe_glass", 5],
       ["coke", 8],
+	  [ "hat_golf", 15 ],
```
Comment (Night-Pryanik) [inline data/json/item_groups.json]: Indent here and below.

File: data/json/items/armor.json [hunk 2]
```diff
@@ -9666,5 +9666,64 @@
...
+    "warmth": 5,
+    "material_thickness": 1,
+    "environmental_protection": 1,
+    "flags": [ "SUN_GLASSES" ]
+  },
+  {
+    "id": "golf_bag",
+    "type": "ARMOR",
+    "name": "golf_bag",
+    "description": "A tall canvas and plastic bag with fold-out legs used for golfing. It even has straps to be worn on the back.",
```
Comment (Night-Pryanik) [inline data/json/items/armor.json]: Two spaces after periods.

File: data/json/items/armor/boots.json [hunk 3]
```diff
@@ -501,6 +501,26 @@
     "material_thickness": 3,
     "flags": [ "VARSIZE" ]
   },
+  {
+    "id": "golf_shoes",
+    "type": "ARMOR",
+    "name": "pair of golf shoes",
+    "name_plural": "pairs of golf shoes",
+    "description": "Snug fit sport shoes with hard ridges on the bottom",
```
Comment (Night-Pryanik) [inline data/json/items/armor/boots.json]: Period at the end.

File: data/json/overmap_specials.json [hunk 4]
```diff
@@ -2240,5 +2240,29 @@
...
+  },
+  {
+    "type": "overmap_special",
+    "id": "golf course",
+    "overmaps": [
+      { "point": [ 0, 0, 0 ], "overmap": "golfcourse_00_north" },
+      { "point": [ 0, 1, 0 ], "overmap": "golfcourse_01_north" },
+      { "point": [ 0, 2, 0 ], "overmap": "golfcourse_02_north" },
+      { "point": [ 1, 0, 0 ], "overmap": "golfcourse_10_north" },
+	  { "point": [ 1, 1, 0 ], "overmap": "golfcourse_11_north" },
```
Comment (Night-Pryanik) [inline data/json/overmap_specials.json]: Indent here and below.

File: data/json/vehicles/carts.json [hunk 5]
```diff
@@ -97,9 +97,48 @@
...
+    "blueprint": [
+      [ "o|o" ],
+      [ "o|o" ]
+    ],
+    "parts": [
+      { "x": 0, "y": 0, "part": "frame_horizontal" },
+      { "x": 0, "y": 0, "part": "seat" },
+      { "x": 0, "y": 0, "part": "roof" },
+      { "x": 0, "y": 0, "part": "engine_electric" },
+      { "x": 0, "y": 0, "part": "controls" },
```
Comment (Night-Pryanik) [inline data/json/vehicles/carts.json]: Add a dashboard, or one wouldn't be able to control electrics of the car.

Comment (cosmiccoincidence) [reply data/json/vehicles/carts.json]: I tested both the original and my new golf cart and both operate just fine. Also I made the other changes.

**Comment by cosmiccoincidence** [comment]
- No need to add items to clothing and sporting stores as I added them to groups such as "allclothes", "hatstore_hats", "shoestore_shoes", "shoes", "livingroom", "sports", and "allsporting" which is what those locations use.

- Made golf bag a sheath for golf clubs.

-Fixed Pryanik's suggestions.

**Comment by cosmiccoincidence** [comment]
Had to tweak that golf bag sheath a bit. But its finished now.

File: data/json/construction.json [hunk 6]
```diff
@@ -2096,13 +2096,58 @@
         "post_terrain" : "t_dock"
     },{
         "type" : "construction",
-        "description" : "Cut Down Grass",
+        "description" : "Cut Grass",
+        "category" : "OTHER",
+        "required_skills" : [ [ "survival", 0 ] ],
+        "time" : 6,
+        "qualities": [ [ { "id": "CUT", "level": 1 } ] ],
```
[the remark this answers is not in the document]
Comment (cosmiccoincidence) [reply data/json/construction.json]: This was actually how it was before I made my addition (check out your local version of construction.json. But I am willing to change it for the better.

**Comment by cosmiccoincidence** [comment]
@Night-Pryanik 
Let me know what you think of the fix.

File: data/json/construction.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (cainiaowu) [reply data/json/construction.json]: Or add a GRASSCUTTING quality and add it to those tools.

[the remark this answers is not in the document]
Comment (cosmiccoincidence) [reply data/json/construction.json]: @cainiaowu I'm not sure where I add new qualities, if I could get pointed in the right direction I'll look into it.

File: data/json/construction.json [hunk 7]
```diff
@@ -2136,7 +2138,7 @@
         "category" : "OTHER",
         "required_skills" : [ [ "survival", 0 ] ],
         "time" : 6,
-        "qualities": [ [ { "id": "CUT", "level": 1 }, { "id": "DIG", "level": 1 } ] ],
+        "qualities": [ [ { { "id": "DIG", "level": 1 } ] ],
```
Comment (Night-Pryanik) [inline data/json/construction.json]: Excessive `{`.

File: data/json/construction.json [hunk 8]
```diff
@@ -2145,7 +2147,7 @@
...
```
Comment (Night-Pryanik) [inline data/json/construction.json]: The same here.

File: data/json/construction.json [hunk 9]
```diff
@@ -2100,16 +2100,18 @@
         "category" : "OTHER",
         "required_skills" : [ [ "survival", 0 ] ],
         "time" : 6,
-        "qualities": [ [ { "id": "CUT", "level": 1 } ] ],
         "pre_terrain" : "t_grass_long",
-        "post_terrain" : "t_grass"
+        "post_terrain" : "t_grass",
+        "tools": [
+        [ [ "scythe", 2 ], [ "sickle", 2 ] ]
```
Comment (Night-Pryanik) [inline data/json/construction.json]: Remove `2`s, as it would try to consume 2 charges of scythe or sickle, and it's good to go.

**Pull request merged**